### PR TITLE
Port SequencerEntrypoint to use ovm solc

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Breaking #475 into several PRs for simplicity. First PR here is porting `OVM_SequencerEntrypoint` to use the l2 compiler.

### CODE REVIEW

**Review by ben-chain — CHANGES_REQUESTED** [review]
This looks good to me!  Only change we need to make is fix the TEMP compiler version.  I'm gonna build a version based off of `master` now that we can use; hopefully @smartcontracts has a good way to invalidate the cache to make sure it updates.

File: packages/contracts/contracts/optimistic-ethereum/OVM/predeploys/OVM_SequencerEntrypoint.sol [hunk 1]
```diff
@@ -59,43 +63,56 @@ contract OVM_SequencerEntrypoint {
...
+        // Grab the chain ID for the current network.
+        uint256 chainId;
+        assembly {
+            chainId := chainid()
+        }
+
         // Need to decompress and then re-encode the transaction based on the original encoding.
         bytes memory encodedTx = Lib_OVMCodec.encodeEIP155Transaction(
-            Lib_OVMCodec.decompressEIP155Transaction(compressedTx),
+            Lib_OVMCodec.decompressEIP155Transaction(
```
Comment (karlfloersch) [inline packages/contracts/contracts/optimistic-ethereum/OVM/predeploys/OVM_SequencerEntrypoint.sol:74]: You made this stateless & pass in the chainID based on the chainid opcode above. That makes sense! And definitely is a lot clearer than our other method!

File: packages/contracts/test/contracts/OVM/precompiles/OVM_SequencerEntrypoint.spec.ts [hunk 2]
```diff
@@ -2,11 +2,13 @@ import { expect } from '../../../setup'
 
 /* External Imports */
 import { waffle, ethers } from 'hardhat'
-import { ContractFactory, Wallet, Contract } from 'ethers'
+import { ContractFactory, Wallet, Contract, BigNumber } from 'ethers'
 import { smockit, MockContract } from '@eth-optimism/smock'
+import { fromHexString, toHexString } from '@eth-optimism/core-utils'
+import { ecrecover, publicToAddress } from 'ethereumjs-util'
```
Comment (karlfloersch) [inline packages/contracts/test/contracts/OVM/precompiles/OVM_SequencerEntrypoint.spec.ts:8]: Is it possible to use https://docs.ethers.io/v5/api/utils/address/#utils-recoverAddress in favor of adding a new dependency?

**Review by ben-chain — APPROVED** [review]
With the temp removed, LGTM!  ( I reviewed this more extensively in #475 and feedback was addressed :) )

**Review by karlfloersch — COMMENTED** [review]
This looks great! I noticed there might be a dependency that we can remove but overall this looks great!

Going to comb over the tests in a little more detail and follow up with more review

**Review by karlfloersch — CHANGES_REQUESTED** [review]
Ah @smartcontracts don't forget to add a changeset

File: packages/contracts/test/contracts/OVM/precompiles/OVM_SequencerEntrypoint.spec.ts (hunk 2 above)
Comment (smartcontracts) [reply packages/contracts/test/contracts/OVM/precompiles/OVM_SequencerEntrypoint.spec.ts]: Done

**Review by karlfloersch — APPROVED** [review]
Dope!

**Pull request merged**